packages/rum-core/src/domain/resource/resourceUtils.spec.ts

```diff
@@ -313,3 +316,47 @@ describe('shouldTrackResource', () => {
     expect(isAllowedRequestUrl(configuration, 'https://my-domain.com/hello?a=b')).toBe(true)
   })
 })
+
+describe('isDataUrl and sanitizeDataUrl', () => {
+  const longString = new Array(MAX_ATTRIBUTE_VALUE_CHAR_LENGTH).join('a')
+  it('returns truncated url when detects data url of json', () => {
+    const longDataUrl = `data:text/json; charset=utf-8,${longString}`
+
+    const expectedUrl = 'data:text/json; charset=utf-8,'
+    expect(isDataUrlTooLong(longDataUrl)).toEqual(true)
+    expect(sanitizeDataUrl(longDataUrl)).toEqual(expectedUrl)
+  })
+
+  it('returns truncated url when detects data url of html', () => {
+    const longDataUrl = `data:text/html,${longString}`
+    expect(isDataUrlTooLong(longDataUrl)).toEqual(true)
+    expect(sanitizeDataUrl(longDataUrl)).toEqual('data:text/html,')
+  })
+
+  it('returns truncated url when detects data url of image', () => {
+    const longDataUrl = `data:image/svg+xml;base64,${longString}`
+    expect(isDataUrlTooLong(longDataUrl)).toEqual(true)
+    expect(sanitizeDataUrl(longDataUrl)).toEqual('data:image/svg+xml;base64,')
+  })
+  it('returns truncated url when detects plain data url', () => {
+    const plainDataUrl = `data:,${longString}`
+    expect(isDataUrlTooLong(plainDataUrl)).toEqual(true)
+    expect(sanitizeDataUrl(plainDataUrl)).toEqual('data:,')
+  })
+
+  it('returns truncated url when detects data url with exotic mime type', () => {
+    const exoticTypeDataUrl = `data:application/vnd.openxmlformats;fileName=officedocument.presentationxml;base64,${longString}`
+    expect(isDataUrlTooLong(exoticTypeDataUrl)).toEqual(true)
+    expect(sanitizeDataUrl(exoticTypeDataUrl)).toEqual(
+      'data:application/vnd.openxmlformats;fileName=officedocument.presentationxml;base64,'
+    )
+  })
+
+  it('returns the original url when the data url is within limit', () => {
+    expect(isDataUrlTooLong(`data:,${longString.substring(5)}`)).toEqual(false)
+  })
+
+  it('returns null when no data url found', () => {
```

Comment:
<!-- Nitpicks are small, trivial, but necessary changes. Distinguishing nitpick comments significantly helps direct the reader's attention to comments requiring more involvement.  -->
**🥜 nitpick:** ‏ 
```suggestion
  it('returns false when no data url found', () => {
```